js/ui/diagram/ui.diagram.js

```diff
@@ -49,11 +50,81 @@ class Diagram extends Widget {
         this._createComponent($options, DiagramOptions);
 
         !isServerSide && this._diagramInstance.createDocument($content[0]);
+        this._diagramInstance.onChanged = this._raiseValueChangeAction.bind(this);
     }
     _initDiagram() {
         const { DiagramControl } = getDiagram();
         this._diagramInstance = new DiagramControl();
     }
+
+    _getDefaultOptions() {
+        return extend(super._getDefaultOptions(), {
+            /**
+            * @name dxDiagramOptions.onValueChanged
+            * @extends Action
+            * @type function(e)
+            * @type_function_param1 e:object
+            * @type_function_param1_field4 value:object
+            * @type_function_param1_field5 previousValue:object
+            * @action
+            */
+            onValueChanged: null,
+        });
+    }
+
+    /**
+    * @name dxDiagramMethods.getValue
+    * @publicName getValue()
+    * @return string
+    */
+    getValue() {
+        var value;
+        const { DiagramCommand } = getDiagram();
+        this._diagramInstance.commandManager.getCommand(DiagramCommand.Export).execute(function(data) { value = data; });
+        return value;
+    }
+    /**
+    * @name dxDiagramMethods.setValue
+    * @publicName setValue(value)
+    * @param1 value:string
+    */
+    setValue(value) {
+        const { DiagramCommand } = getDiagram();
+        this._diagramInstance.commandManager.getCommand(DiagramCommand.Import).execute(value);
+        this._raiseValueChangeAction();
+    }
+
+    _createValueChangeAction() {
+        this._valueChangeAction = this._createActionByOption("onValueChanged");
+    }
+    _raiseValueChangeAction() {
+        if(!this.option("onValueChanged")) return;
```

Comment:
If widget has the `onSomeAction` option it has the `someAction` event.
So there is 2 ways to run an action handler:
- use option `dxWidget({ onSomeAction: handler })`
- subscribe on event via the `on` method `instance.on("someAction", handler)`